iml-agent/src/action_plugins/lustre/client.rs

```diff
@@ -96,12 +96,16 @@ pub async fn mount_many(xs: Vec<Mount>) -> Result<(), ImlAgentError> {
 pub async fn unmount(unmount: Unmount) -> Result<(), ImlAgentError> {
     delete_fstab_entry(&unmount.mountpoint).await?;
 
-    Command::new("/bin/umount")
-        .arg(unmount.mountpoint)
-        .checked_output()
-        .err_into()
-        .await
-        .map(drop)
+    if is_mounted(&unmount.mountpoint, false).await? {
```

Comment:
Why?

Reply:
So it's idempotent

Reply:
Also masks the error where you ask something to unmount that's already unmounted.

Reply:
Should that be an error?

Reply:
I think so

```
[root@mds1 ~]# umount /foo/bar
umount: /foo/bar: mountpoint not found
[root@mds1 ~]# echo $?
32
```

Reply:
If something fails, it becomes harder to unwedge.

Reply:
This also makes it consistent with `mount()` as calling mount twice doesn't result in double mounting the target.